### ISSUE TITLE
Bad IDE0028 suggestion

### ISSUE THREAD

**Issue by nvmkpk** [dotnet/roslyn#72701]
**Version Used**: 
Visual Studio 17.10.0 Preview 2.0

**Steps to Reproduce**:

Add code like below:
```
IList<string> strings = new ObservableCollection<string>();
```
**Diagnostic Id**: `IDE0028`

**Expected Behavior**:
No `IDE0028` is reported.

**Actual Behavior**:
`IDE0028` is reported and if suggestion is applied, it changes the code to as shown below:
```
IList<string> strings = new [];
```
Compiler generates below code for:
```
IList<string> strings = new List<string>();
```
This changes the application functionality in MVVM.

**Comment by nvmkpk** [comment]
Does this change fix only the usages of `ObservableCollection<>`? Does it handle the case when I define my own implementation of a collection with `ObservableCollection<>` as base class or with `INotifyCollectionChanged` interface implementation?
